velox/exec/fuzzer/AggregationFuzzer.cpp

```diff
@@ -684,22 +737,85 @@ void resetCustomVerifiers(
   }
 }
 
+bool AggregationFuzzer::compareEquivalentPlanResults(
+    const core::PlanNodePtr& plan,
+    bool customVerification,
+    const std::vector<RowVectorPtr>& input,
+    const std::vector<PlanWithSplits>& plans,
+    const std::vector<std::shared_ptr<ResultVerifier>>& customVerifiers,
+    int32_t maxDrivers,
+    bool distinct) {
+  try {
+    auto resultOrError = execute(plan);
+    if (resultOrError.exceptionPtr) {
+      ++stats_.numFailed;
+    }
+
+    // TODO Use ResultVerifier::compare API to compare Velox results with
+    // reference DB results once reference query runner is updated to return
+    // results as Velox vectors.
+    std::optional<MaterializedRowMultiset> expectedResult;
+    if (resultOrError.result != nullptr) {
+      if (!customVerification) {
+        auto referenceResult = computeReferenceResults(plan, input);
+        updateReferenceQueryStats(referenceResult.second);
+        expectedResult = referenceResult.first;
+      } else {
+        ++stats_.numVerificationSkipped;
+
+        for (auto& verifier : customVerifiers) {
+          if (verifier != nullptr && verifier->supportsVerify()) {
+            VELOX_CHECK(
+                verifier->verify(resultOrError.result),
+                "Aggregation results failed custom verification");
+          }
+        }
+      }
+    }
+
+    if (expectedResult && resultOrError.result) {
+      ++stats_.numVerified;
+      VELOX_CHECK(
+          assertEqualResults(
+              expectedResult.value(),
+              plan->outputType(),
+              {resultOrError.result}),
+          "Velox and reference DB results don't match");
+      LOG(INFO) << "Verified results against reference DB";
+    }
+
+    testPlans(
+        plans,
+        customVerification,
+        customVerifiers,
+        resultOrError,
+        maxDrivers,
+        distinct);
+
+    return resultOrError.exceptionPtr != nullptr;
+  } catch (...) {
+    if (!reproPersistPath_.empty()) {
+      persistReproInfo(plans, reproPersistPath_);
```

Comment:
Could you run an experiment to confirm whether persistReproInfo and velox_aggregation_runner_test honor distinct aggregation plans?